### PR TITLE
Fix Linting Errors

### PR DESCRIPTION
**Pull request by devopstagon**
## Description

Fixed linting errors. I also found a possible slowloris error and fixed that as well:

https://github.com/devopstagon/oauth2-proxy/blob/master/pkg/http/server.go#L189
https://medium.com/a-journey-with-go/go-understand-and-mitigate-slowloris-attack-711c1b1403f6

## Motivation and Context

I wanted to try and work on the Azure provider a bit, but I figured it would be a good idea to preemptively merge this into master first to have a "clean" base to work from. 

## How Has This Been Tested?

I ran make and also built an image and tested it in my environment.

## Checklist:

- [no] My change requires a change to the documentation or CHANGELOG.
- [no] I have updated the documentation/CHANGELOG accordingly.
- [no] I have created a feature (non-master) branch for my PR.

### CODE REVIEW

**Comment by devopstagon** [comment]
Also fixes: https://github.com/oauth2-proxy/oauth2-proxy/issues/1646

**Comment by devopstagon** [comment]
When it says make a non master branch does it mean on my fork?

**Comment by JoelSpeed** [comment]
> When it says make a non master branch does it mean on my fork?

Yes, typically we work from feature branches. There can sometimes be issues with the way our builds run if you PR from a branch called master, though I think they have been resolved now

File: pkg/http/server.go [hunk 1]
```diff
@@ -186,7 +186,7 @@ func (s *server) Start(ctx context.Context) error {
 // When the given context is cancelled the server will be shutdown.
 // If any errors occur, only the first error will be returned.
 func (s *server) startServer(ctx context.Context, listener net.Listener) error {
-	srv := &http.Server{Handler: s.handler}
+	srv := &http.Server{Handler: s.handler, ReadHeaderTimeout: time.Minute}
```
Comment (JoelSpeed) [inline pkg/http/server.go:189]: Where did this one come from?

File: pkg/requests/result.go [hunk 2]
```diff
@@ -66,10 +66,10 @@ func (r *result) UnmarshalInto(into interface{}) error {
 	return nil
 }
 
-// UnmarshalJSON performs the request and attempts to unmarshal the response into a
+// UnmarshalSimpleJSON performs the request and attempts to unmarshal the response into a
```
Comment (JoelSpeed) [inline pkg/requests/result.go:69]: Good work, thanks!

**Review by JoelSpeed — COMMENTED** [review]
Quick question about one of the changes which looks like it could be a behaviour change, otherwise LGTM

File: pkg/http/server.go (hunk 1 above)
Comment (devopstagon) [reply pkg/http/server.go:189]: This is to mitigate a potential slowloris attack. https://medium.com/a-journey-with-go/go-understand-and-mitigate-slowloris-attack-711c1b1403f6

I suppose we could make it configurable, but having this unset leaves the server vulnerable.

**Comment by adriananeci** [comment]
Closing as duplicate of https://github.com/oauth2-proxy/oauth2-proxy/pull/1835

**Review by JoelSpeed — APPROVED** [review]

**Pull request merged**